classes/task/FileLoader.php

```diff
@@ -269,6 +284,13 @@ protected function executeActions(): bool
                 // Let the script know what files were sent back to staging,
                 // so it doesn't claim them again thereby entering an infinite loop.
                 $this->_stagedBackFiles[] = $this->_claimedFilename;
```

Comment:
Unrelated to the changes, but if you replace the `opendir()` by the `FilesystemIterator`, then this won't be needed.